### PR TITLE
EnMu header and pointer fixes

### PR DESCRIPTION
**Pull request by liamnobel**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Mu/z_en_mu.c [hunk 1]
```diff
@@ -102,7 +108,8 @@ u16 EnMu_GetFaceReaction(GlobalContext* globalCtx, Actor* thisx) {
     return this->defFaceReaction;
 }
 
-s32 EnMu_CheckDialogState(GlobalContext* globalCtx, EnMu* this) {
+s16 EnMu_CheckDialogState(GlobalContext* globalCtx, Actor* thisx) {
+    EnMu* this = THIS;
     switch (func_8010BDBC(&globalCtx->msgCtx)) {
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Mu/z_en_mu.c:113]:
```suggestion
    EnMu* this = THIS;
    
    switch (func_8010BDBC(&globalCtx->msgCtx)) {
```

File: src/overlays/actors/ovl_En_Mu/z_en_mu.c [hunk 2]
```diff
@@ -159,12 +165,12 @@ void EnMu_Update(Actor* thisx, GlobalContext* globalCtx) {
...
     this->collider.dim.pos = pos;
 
-    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
     SkelAnime_Update(&this->skelAnime);
     Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
     this->actionFunc(this, globalCtx);
     talkDist = this->collider.dim.radius + 30.0f;
-    func_800343CC(globalCtx, this, &this->dialogState, talkDist, EnMu_GetFaceReaction, EnMu_CheckDialogState);
+    func_800343CC(globalCtx, thisx, &this->dialogState.unk_00, talkDist, EnMu_GetFaceReaction, EnMu_CheckDialogState);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Mu/z_en_mu.c:173]:
```suggestion
    func_800343CC(globalCtx, &this->actor, &this->dialogState.unk_00, talkDist, EnMu_GetFaceReaction, EnMu_CheckDialogState);
```
If this doesn't work than change `this->actor` to `thisx`

**Review by louist103 — CHANGES_REQUESTED** [review]
It won't let me add comments on the following things that need to be fixed:
Line break on line 180, and 194
Remove line break from 202

**Comment by Zelllll** [comment]
The name `dialogState` is not accurate. It should be called `npcInfo`, as that is what that struct handles.

**Review by louist103 — APPROVED** [review]

**Review by Zelllll — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]

**Pull request merged**